---
title: "CCLE_TCGA_Analysis"
output: html_document
#Code originally generated by Dianne Lumaquin: Lumaquin-Yin, et al., 2023
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Load required packages
```{r packages}
library(dplyr)
library(Matrix)
library(gplots)
library(ggplot2)
library(tidyverse)
library(scales)
library(pals)
library(cowplot)
library(fgsea)
library(msigdbr)
library(data.table)
library(conflicted)
conflict_prefer("filter", "dplyr")
conflict_prefer("desc", "dplyr")
conflict_prefer("slice", "dplyr")
library(readxl)
library(tidyr)
library(viridis)
library(gridExtra)
library(ComplexHeatmap)
library(ggpubr)
conflict_prefer("viridis", "pals")

```


#Celligner classification for TCGA
```{r celligner}
celligner <- read.csv("celligner_alignment.csv")
head(celligner)
celligner <- tibble(celligner)
celligner <- select(celligner, -X)
```

#Load metadata for TCGA
```{r tcga}
tcga_cell <- celligner %>% dplyr::filter(type == "tcgaplus-tumor")
tcga_cell <- tcga_cell %>% dplyr::filter(lineage == "skin")
tcga_cell_mel <- tcga_cell %>% dplyr::filter(subtype == "melanocytic")
tcga_cell_mel$tsoi_class <- "melanocytic"

tcga_cell_un <- tcga_cell %>% dplyr::filter(subtype == "undifferentiated")
tcga_cell_un$tsoi_class <- "undifferentiated"
tcga_cell_nc <- tcga_cell %>% dplyr::filter(subtype == "neural crest-like")
tcga_cell_nc$tsoi_class <- "neural crest-like"
aa <- tcga_cell %>% dplyr::filter(subtype == "transitory")
aa$tsoi_class <- "transitory"
tcga_meta <- full_join(tcga_cell_un, tcga_cell_nc)
tcga_meta <- full_join(tcga_meta, aa)


tcga_meta <- full_join(tcga_meta, tcga_cell_mel)
tcga_meta
```

#Load TCGA expression data
```{r}
genes_tcga <- read.csv("genes_tcga_Dianneedits.csv")

```

#merge metadata and expression data
```{r make exp_TCGA table}
meta_TCGA <- tcga_meta %>% select(sampleID, subtype, tsoi_class)
meta_TCGA
head(meta_TCGA)
meta_TCGA

exp_tcga <- left_join(meta_TCGA, genes_tcga)
dim(exp_tcga)

#genes with classification only
genes_tcga_cut <- right_join(genes_tcga, exp_tcga)
genes_tcga_cut

```


#Plot individual genes of interest
```{r}
#gene of interest
DHRS3 <- ggplot(exp_tcga, aes(x = factor(tsoi_class, levels = c("undifferentiated", "neural crest-like", "transitory", "melanocytic")), y = DHRS3, fill = factor(tsoi_class, levels = c("undifferentiated", "neural crest-like", "transitory", "melanocytic")))) + 
  geom_violin() +
  stat_summary(fun='median', geom='crossbar', size= 0.3 , col='black') +
  geom_jitter(shape=16) +
  theme_bw() +
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
panel.grid.minor = element_blank(), axis.line = element_line(colour = "black", size = 1)) +
   scale_fill_manual(values=c("#440154", "#29778E", "#404386", "#3CBB75")) +
  theme(axis.text.x = element_text(angle = 10, vjust = 0.5)) +
  xlab(NULL) + 
  ylab("log2(normalized expression)") +
  theme(plot.title = element_text(hjust = 0.5), legend.position="none")+
  theme(text = element_text(size = 26))
DHRS3

ggsave("DHRS3_exp_TCGA_4.pdf", plot = DHRS3, dpi = "retina", width = 7, height = 7)
ggsave("DHRS3_exp_TCGA_4.svg", plot = DHRS3)

```


#Stats for violin plots
```{r stats}
pairwise.wilcox.test(exp_tcga$DHRS3, meta_TCGA$tsoi_class,
                 p.adjust.method = "holm")

```


#CCLE analysis
Gene expression TPM values of the protein coding genes for DepMap cell lines. 
Values are inferred from RNA-seq data using the RSEM tool and are reported after log2 transformation, using a pseudo-count of 1; log2(TPM+1).

```{r read data}
CCLE <- read.csv("CCLE_expression_revised.csv")
CCLE <- CCLE %>% remove_rownames %>% column_to_rownames(var="X")
CCLE <- CCLE %>%
   tibble::rownames_to_column() %>%  
   pivot_longer(-rowname) %>% 
   pivot_wider(names_from=rowname, values_from=value) 
CCLE <- tibble(CCLE)
CCLE <- CCLE %>% column_to_rownames(var = "name")
head(CCLE)
```


#Celligner cell lines
```{r cell lines}


celligner <- read.csv("celligner_alignment.csv")
celligner_mel <- celligner %>% filter(type == "depmap-model")
celligner_mel <- celligner_mel %>% filter(lineage == "skin")

x <- celligner_mel %>% filter(subtype == "undifferentiated")
y <- celligner_mel %>% filter(subtype == "neural crest-like")
z <- celligner_mel %>% filter(subtype == "transitory")
z
#celligner_nu <- full_join(x, y)
#celligner_nu$tsoi_class <- "neural crest/undifferentated"
celligner_melano <- celligner_mel %>% filter(subtype == "melanocytic")
celligner_melano$tsoi_class <- "melanocytic"
celligner_trans <- z
celligner_trans$tsoi_class <- "transitory"
celligner_un <- x
celligner_un$tsoi_class <- "undifferentiated"
celligner_nc <- y
celligner_nc$tsoi_class <- "neural crest-like"


celligner_melano

celligner_master <- full_join(celligner_un, celligner_trans)
celligner_master <- full_join(celligner_master, celligner_nc)
celligner_master <- full_join(celligner_master, celligner_melano)
celligner_master

celligner_un_names <- celligner_un$sampleID
celligner_nc_names <- celligner_nc$sampleID
celligner_melano_names <- celligner_melano$sampleID
celligner_trans_names <- celligner_trans$sampleID
```

```{r compare gene expression}
mel_CCLE <- CCLE %>% dplyr::select(all_of(celligner_melano_names))
mel_CCLE <- mel_CCLE %>% rownames_to_column(var = "Gene")
mel_CCLE

un_CCLE <- CCLE %>% dplyr::select(all_of(celligner_un_names))
un_CCLE <- un_CCLE %>% rownames_to_column(var = "Gene")
un_CCLE

nc_CCLE <- CCLE %>% dplyr::select(all_of(celligner_nc_names))
nc_CCLE <- nc_CCLE %>% rownames_to_column(var = "Gene")
nc_CCLE

trans_CCLE <- CCLE %>% dplyr::select(all_of(celligner_trans_names))
trans_CCLE <- trans_CCLE %>% rownames_to_column(var = "Gene")
trans_CCLE

all_CCLE <- full_join(un_CCLE, trans_CCLE)
all_CCLE <- full_join(all_CCLE, nc_CCLE)
all_CCLE <- full_join(all_CCLE, mel_CCLE)
head(all_CCLE)
```

```{r make exp_CCLE table}
meta_CCLE <- celligner_master %>% select(sampleID, display_name, subtype, tsoi_class)
meta_CCLE

#fixing gene names
ab <- all_CCLE
ab.gene <- ab$Gene
ab.gene <- gsub("\\..[0-9].*","", ab.gene)
ab.gene
ab$gene.name <- ab.gene
ab <- distinct(ab, gene.name, .keep_all= TRUE)
ab$Gene <- NULL
ab


genes_CCLE <- ab %>%
  pivot_longer(cols = -gene.name) %>% 
  pivot_wider(names_from = gene.name, values_from = value)
genes_CCLE
genes_CCLE <- rename(genes_CCLE, sampleID = name)

head(genes_CCLE)

exp_CCLE <- full_join(meta_CCLE, genes_CCLE)
exp_CCLE

write.csv(exp_CCLE, "exp_CCLE.csv")
```

#Plot individual genes of interest
```{r}
DHRS3 <- ggplot(exp_CCLE, aes(x = factor(tsoi_class, levels = c("undifferentiated", "neural crest-like", "transitory", "melanocytic")), y = DHRS3, fill = factor(tsoi_class, levels = c("undifferentiated", "neural crest-like", "transitory", "melanocytic"))))  + 
  geom_violin() +
   stat_summary(fun='median', geom='crossbar', size= 0.3 , col='black') +
  geom_jitter(shape=16) +
  theme_bw() +
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
panel.grid.minor = element_blank(), axis.line = element_line(colour = "black", size = 1)) +
   scale_fill_manual(values=c("#440154", "#29778E", "#404386", "#3CBB75")) +
  theme(axis.text.x = element_text(angle = 10, vjust = 0.5)) +
  xlab(NULL) + 
  ylab("log2(normalized expression)") +
  theme(plot.title = element_text(hjust = 0.5), legend.position="none")+
  theme(text = element_text(size = 26))
DHRS3

ggsave(filename = "DHRS3_exp_CCLE_4.pdf", plot = DHRS3, dpi = "retina")
ggsave(filename = "DHRS3_exp_CCLE_4.svg", plot = DHRS3)

```


#statistics
```{r}
pairwise.wilcox.test(exp_CCLE$DHRS3, exp_CCLE$tsoi_class,
                 p.adjust.method = "holm")

```

#heatmap of tsoi signature genes for each cell line
```{r pigment heatmap}

t_sig <- read_csv("tsoi_signature.csv")
t_sig <- c("WNT5A", "TWIST2", "SOX9", "AXL", "GLI2", "SOX10", "TYR", "DCT", "TYRP1", "MITF", "PMEL", "SLC45A2")
t_sig <- data_frame(t_sig)

t_sig_exp <- ab %>% filter(ab$gene.name %in% t_sig$t_sig)
t_sig_exp <- t_sig_exp %>% column_to_rownames(var = "gene.name")
order_tsoi <- exp_CCLE$sampleID
t_sig_exp <- select(t_sig_exp, order_tsoi)

listdf <- list(t_sig_exp)
listdf <- lapply(listdf, setNames, exp_CCLE$display_name)
t_sig_exp_named <- as.data.frame(listdf[[1]])
t_sig_exp_named
t_sig_exp_3739 <- select(t_sig_exp_named, IGR37, IGR39)
t_sig_exp_matrix <- data.matrix(t_sig_exp_3739)


library(matrixStats)
z_score_matrix <- t(scale(t(t_sig_exp_matrix)))

#pdf(file = "selectgenes_heatmap_CCLE_IGR37IGR39.pdf")
Heatmap(matrix = t_sig_exp_matrix,
        name = "z score",
        col = viridis(5),
        column_names_side = "top",
        column_names_rot = 90,
        show_row_dend = F,
        show_column_dend = T,
        row_names_gp = gpar(fontsize = 20),
        column_names_gp = gpar(fontsize = 20),
        cluster_rows = TRUE,
        show_row_names = TRUE,
        row_names_side = "left",
        #column_split = exp_CCLE$tsoi_class, 
        column_title_rot = 90) 
        #cluster_columns = FALSE,
        #row_split = t_sig_edit$Signature)
#dev.off()


## check this
geneset <- t_sig$Gene
  
check <- data.frame(geneset, geneset %in% names(exp_CCLE)) 
check <- check[check$geneset..in..names.exp_CCLE. != FALSE, ]  

t_sig_edit <- merge(t_sig, check, by.x = 'Gene', by.y = 'geneset')

```

#Plot lipid droplet genes
```{r}


EJMS <- read.csv("/Volumes/whitelab/Lab Members/Eleanor Johns/Lipid Droplet Project/APEX2 Mass Spec/Lipid Droplet Isolations/MassSpec_A375/MassSpecAnalysis/FinalAnalysis/EJ_MSanalysis_PLIN2-Cyto_highconf.csv")

LD_sig <- EJMS[1:30,]

#LD_sig <- LD_sig[!(row.names(LD_sig) %in% "67"),]
LD_sig <- data.frame(LD_sig)
write.csv(LD_sig, "LD_sig.csv")
LD_sig_exp <- ab %>% filter(ab$gene.name %in% LD_sig$LD_sig)
LD_sig_exp <- LD_sig_exp %>% column_to_rownames(var = "gene.name")
order_LD <- exp_CCLE$sampleID
LD_sig_exp <- select(LD_sig_exp, order_LD)

listdf <- list(LD_sig_exp)
listdf <- lapply(listdf, setNames, exp_CCLE$display_name)
LD_sig_exp_named <- as.data.frame(listdf[[1]])
LD_sig_exp_named
LD_sig_exp_3739 <- select(LD_sig_exp_named, IGR37, IGR39)
LD_sig_exp_matrix <- data.matrix(LD_sig_exp_3739)

# Install and load the ggplot2 library
#install.packages("ggplot2")
library(ggplot2)
# Assuming you have your data in a matrix named 'data_matrix'
# Convert the matrix to a dataframe for ggplot
df <- as.data.frame(LD_sig_exp_matrix)

#Correlation plot

library(ggplot2)
# Create the scatter plot
ggplot(df, aes(x=IGR37, y=IGR39)) +
  geom_point() +
  geom_text_repel(data = df[-grep("DHRS3", rownames(df)),],
                                 label=rownames(df[-grep("DHRS3", rownames (df)),]), 
                  size = 8, point.padding = 1, segment.color = 'grey50', max.overlaps = 15) +
  geom_abline(intercept=0, slope=1, color="black", linetype="dashed") +
  ggrepel::geom_text_repel(data =  df[grep("DHRS3", rownames(df)),], #label differently DHRS3
                           label = rownames(df[grep("DHRS3", rownames(df)),]), 
                           size = 12,
                           fontface = "bold",
                           color= "#29778E",
                           nudge_x = 0.2, 
                           nudge_y = 0.1)+
  ggtitle("Gene Expression Comparison: IGR37 vs. IGR39") +
  xlab("Expression in IGR37") +
  ylab("Expression in IGR39") +
  theme_minimal(base_size = 36) +
ggpubr::stat_cor(method = "spearman", label.x = -1.5, label.y = 10, size = 10)+ # add the correlation coefficient and the p-value (i've put it in an empty place)
  ggplot2::geom_point(data = df[grep("DHRS3", rownames(df)),], color = "#29778E", size  = 4) # change the color and size of DHRS3 

ggsave("LDgeneexp_IGR37OGR39.pdf", dpi=300)
ggsave("LDgeneexp_IGR37IGR39.svg")

```


#CCLE Proteomics Analysis: Jeremy Raymond

```{r}
#Load the data 
Melanoma_proteomics <- readxl::read_excel("/Volumes/whitelab/Lab Members/Eleanor Johns/Lipid Droplet Project/APEX2 Mass Spec/Lipid Droplet Isolations/MassSpec_A375/MassSpecAnalysis/FinalAnalysis/CCLEandTCGAexp/DepMap_proteomics_Jeremy/Melanoma_CCLE_Proteomics.xlsx") #Load the proteomic data 
CL_annotations <- readxl::read_excel("/Volumes/whitelab/Lab Members/Eleanor Johns/Lipid Droplet Project/APEX2 Mass Spec/Lipid Droplet Isolations/MassSpec_A375/MassSpecAnalysis/FinalAnalysis/CCLEandTCGAexp/DepMap_proteomics_Jeremy/celligner_alignment.xlsx") #Load the annotation data
LD_proteins <- readxl::read_excel("/Volumes/whitelab/Lab Members/Eleanor Johns/Lipid Droplet Project/APEX2 Mass Spec/Lipid Droplet Isolations/MassSpec_A375/MassSpecAnalysis/FinalAnalysis/CCLEandTCGAexp/DepMap_proteomics_Jeremy/LD_sig.xlsx") #Load the list of LD proteins



########## CORRELATION GRAPH ##########  

#Fetch the IDs for the IGR37 and IGR39
IGR_Proteomics <- as.data.frame(subset(Melanoma_proteomics,cell_line_display_name %in% c("IGR39", "IGR37")))

#Set cell names as row names
rownames(IGR_Proteomics) <- IGR_Proteomics$cell_line_display_name

#Delete the uniprot IDs from the protein names
colnames(IGR_Proteomics) <- sub(" .*", "", colnames(IGR_Proteomics))

#subset the table to only the proteins of interest + we transpose the table for easier use
IGR_LDs <- as.data.frame(t(IGR_Proteomics[,LD_proteins$LD_sig]))

#Set protein without expression to -2 (low value to keep the data plotted)
#Not mandatory and can be adjusted (I chose -2 as it's lower than every other value)
IGR_LDs[is.na(IGR_LDs)] <- -2


## data are ready to be plotted ##

```

```{r}
library(ggplot2)
# Create the scatter plot
p <- ggplot(IGR_LDs, aes(x=IGR37, y=IGR39)) +
  geom_point() +
  geom_text_repel(data = IGR_LDs[-grep("DHRS3", rownames(IGR_LDs)),],
                                 label=rownames(IGR_LDs[-grep("DHRS3", rownames (IGR_LDs)),]),
                  size = 8, point.padding = 1, segment.color = 'grey50') +
  geom_abline(intercept=0, slope=1, color="black", linetype="dashed") +
  ggrepel::geom_text_repel(data =  IGR_LDs[grep("DHRS3", rownames(IGR_LDs)),], #label differently DHRS3
                           label = rownames(IGR_LDs[grep("DHRS3", rownames(IGR_LDs)),]), 
                           size = 12,
                           fontface = "bold",
                           color= "#29778E",
                           nudge_x = 0.2, 
                           nudge_y = 0.1)+
  ggtitle("Proteomics Comparison: IGR37 vs. IGR39") +
  xlab("ZScore in IGR37") +
  ylab("ZScore in IGR39") +
  theme_minimal(base_size = 36) +
ggpubr::stat_cor(method = "spearman", label.x = -1.5, label.y = 2, size = 10)+ # add the correlation coefficient and the p-value (i've put it in an empty place)
  ggplot2::geom_point(data = IGR_LDs[grep("DHRS3", rownames(IGR_LDs)),], color = "#29778E", size  = 4) # change the color and size of DHRS3 
p
#To save the plot, you can adapt each parameter as you which. 
#I recommend to save it as a pdf as you can further modify the graph in illustrator 
ggplot2::ggsave(plot = p, filename = "Proteomics_LDcorrelation_IGR37IGR39.pdf",
       path = "/Volumes/whitelab/Lab Members/Eleanor Johns/Lipid Droplet Project/APEX2 Mass Spec/Lipid Droplet Isolations/MassSpec_A375/MassSpecAnalysis/FinalAnalysis/CCLEandTCGAexp/DepMap_proteomics_Jeremy",
       dpi = 300,
       device = "pdf")
ggsave(filename = "Proteomics_LDcorrelation_IGR37IGR39.svg", plot = p, path = "/Volumes/whitelab/Lab Members/Eleanor Johns/Lipid Droplet Project/APEX2 Mass Spec/Lipid Droplet Isolations/MassSpec_A375/MassSpecAnalysis/FinalAnalysis/CCLEandTCGAexp/DepMap_proteomics_Jeremy")

```